lib/table_of_contents/parser.rb

```diff
@@ -40,29 +42,20 @@ def toc
       # parse logic is from html-pipeline toc_filter
       # https://github.com/jch/html-pipeline/blob/v1.1.0/lib/html/pipeline/toc_filter.rb
       def parse_content
-        entries = []
         headers = Hash.new(0)
 
-        if @ignore_within
-          @doc.css(@ignore_within).remove
-        end
-
-        # TODO: Use kramdown auto ids
-        @doc.css(toc_headings).reject { |n| n.classes.include?('no_toc') }.each do |node|
+        @doc.css(".#{@no_toc_section_class}").remove if @no_toc_section_class
```

Comment:
This behavior is not expected because the original nodes inside `@no_toc_section_class` is lost. I need to find a way to filter `@no_toc_section_class` without loosing any contents.